### ISSUE TITLE
[Feat] Support for any file type

### ISSUE THREAD

**Issue by KebPericles** [OneFolderApp/OneFolder#78]
I like the idea, but I would like to be able to use tags in any file, not just images, is there any limitation for that or its just that it doesn't align with the project objective? 

Also if there is no limitation I would like to make it possible rather contributing to this project or forking it, but I have no idea of where to start, so an small hint to start with it would be nice.

Or maybe you know a similar project that has this objective, I would love to take a look in them.

**Comment by Antoine-lb** [comment]
> I like the idea, but I would like to be able to use tags in any file, not just images, is there any limitation for that or its just that it doesn't align with the project objective?

Yes it's something we would like to have. The reazons on why we don't have it yet is:
- no time to develop it
- only a hand full of file format support metadata tags, so it means breaking the original promise where all the data it's stored in the file itself, but we are ready to do that, but:
- We need to re-imagine the app to work more in hand with a database, more on this later.

> Also if there is no limitation I would like to make it possible rather contributing to this project or forking it, but I have no idea of where to start, so an small hint to start with it would be nice.

We are more than open for big contributions, so much so we can have a call if you want to be onboarded

> Or maybe you know a similar project that has this objective, I would love to take a look in them.

Maybe [TagStudio](https://github.com/TagStudioDev/TagStudio) or [Eagle](https://en.eagle.cool/) could do the job? if you have one that does the job I would like to know about it.

**Comment by Antoine-lb** [comment]
So more in the dev side, I think there is 3 things that need to be addressed:
- indexing
- database
- thumbnails and preview

### Indexing
The Allusion team did a greate job at this, and I think this should not be that hard. We just have to read all files and throw them in the database.

### Database
Like I said in the other comment, very few files have metadata support (JPGs, PNGs and other media files). But most don't, this means that now the data in the database is important and needs to be backed-up.

Same as before, the Allusion team did great work on this, maybe changing some stuff from here and there, but mostly it should be fine because the have auto-backups and everything.

But, I would like the database to be stored with the files themselvs, and not in a random location. So if I put them on a hard-drive and go away I have the database with me.

### Thumbnails and Preview
Here is the big part, but doesn't need to be done all at once. What happens with that file? how do we display it?

What I mean is that each file format is handeled differently, like how do I show a .ppt file?

Initially just showing a thumnail saying that this file format is not supported but you can still see it there. Then start adding support for new files one by one.

---

Honestly, these should be doable.

I see you have repos in spanish, I speak spanish too, in case you want to have a call.

And thanks for taking interes on the project, lately the project has been a bit abandoned by us bc outside work is taking all of our time, but in 1 or 2 months we should be back pushing new features.

**Comment by Antoine-lb** [comment]
https://onefolder.canny.io/admin/feedback/feedback/p/support-all-the-files-format-to-replace-finder
